### PR TITLE
us65 acceptance tests

### PR DESCRIPTION
**Pull request by MandyNimmo**
acceptance tests for us65 delete borrower personal info

### CODE REVIEW

File: features/step_definitions/generic_api.rb [hunk 1]
```diff
@@ -25,3 +25,11 @@
 And(/^a message for failure is given "([^"]*)"$/) do |failure_reason|
   assert_match(failure_reason, @response)
 end
+
+And(/^I know the borrower id$/) do
+  deed_end_point = @response.body.to_s
+  @response = HTTP.get(deed_end_point, json: @deed_hash)
```
Comment (nigelpain) [inline features/step_definitions/generic_api.rb]: Just a thought that I'm not sure if you need to include the 'json: @deed_hash' bit in the get request as I don't think you need to pass anthing through in the payload with a get request. Might want to just double check with a dev though.

File: features/us65_delete_borrower_personal_info.feature [hunk 2]
```diff
@@ -0,0 +1,19 @@
+@us65_delete_borrower_personal_info
```
Comment (nigelpain) [inline features/us65_delete_borrower_personal_info.feature]: Would be tempted to split this tag up into two seperate tags like it's done in the other feature files, so have one with just the story number @us65 which is nice and quick and easy to use for those in the know. But also the longer one using the story name @delete_borrower_personal_info for those who might not be up to date with the user story number being used.

File: features/us65_delete_borrower_personal_info.feature [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+  And I create the deed via the Deed API
+  And I know the borrower id
+
+Scenario: Delete Single Borrowers Details
+  When I delete the deed via the Deed API
+  Then a status code of "200" is returned
+
+Scenario: Delete Borrowers Details on an invalid Deed ID
+  When I try to delete a borrowers details supplying an invalid Deed ID
+  Then a status code of "404" is returned
```
Comment (nigelpain) [inline features/us65_delete_borrower_personal_info.feature:21]: Just wondering about if this should be a 404 error as seems to suggest this isnt being handled by the code/schema as if so I would have expected a 400 response. Might just want to double check this is correct with dev.

File: features/us65_delete_borrower_personal_info.feature [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+  mortgage application has been registered.
+
+Background:
+  Given I have valid deed data
+  And I create the deed via the Deed API
+  And I know the borrower id
+
+Scenario: Delete Single Borrowers Details
+  When I delete the deed via the Deed API
+  Then a status code of "200" is returned
```
Comment (nigelpain) [inline features/us65_delete_borrower_personal_info.feature:15]: I'm not sure how can we do this but just thinking if it's enough to check that a 200 is returned or do we also want to check that the data has actually been deleted? Might want to double check with dev if this has been covered at unit test level, if so just checking the 200 will be ok.

File: features/step_definitions/generic_api.rb (hunk 1 above)
Comment (MandyNimmo) [inline features/step_definitions/generic_api.rb]: confirmed with Ramin and removed as suggested :-)

File: features/us65_delete_borrower_personal_info.feature (hunk 2 above)
Comment (MandyNimmo) [inline features/us65_delete_borrower_personal_info.feature]: didnt realise you could have more than one tag - amended as suggested

File: features/us65_delete_borrower_personal_info.feature (hunk 3 above)
Comment (MandyNimmo) [inline features/us65_delete_borrower_personal_info.feature:21]: Checked before writing tests and Ramin said it was a 404

File: features/us65_delete_borrower_personal_info.feature (hunk 4 above)
Comment (MandyNimmo) [inline features/us65_delete_borrower_personal_info.feature:15]: It hasnt been covered in unit test, Simon T thought it should be an integration test but Ramin didn't think it was possible. So have added an extra step to try to delete the personal info again with an expected 404 response as it no longer exists

**Comment by nigelpain** [comment]
:+1:

**Pull request merged**